src/request/xhr.ts

```diff
@@ -60,7 +66,7 @@ export default function xhr<T>(url: string, options: XhrRequestOptions = {}): Re
 
 		let timeoutHandle: Handle;
 		request.onreadystatechange = function (): void {
-			if (request.readyState === 4) {
+			if (!isAborted && request.readyState === 4) {
```

Comment:
If the request times out, then the underlying promise is rejected. However, since the `request` object itself is unaware of any promise state, it fires that `onreadystatechange` callback anyway. But you're right; when `Task#cancel` is executed, the task itself is not rejected. So I wonder if rejecting the promise as aborted within the local `abort` function and resetting `onreadystatechange` to a noop function is the way to go here.